automation/release.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -exuo pipefail
+
+tag="$(git tag --points-at HEAD | head -1)"
+if [ -z "$tag" ]; then
+    exit 0
+fi
+
+function cleanup_gh_install() {
+    rm -f "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    rm -rf "gh_${GH_CLI_VERSION}_linux_amd64/"
+}
+
+if ! command -V gh; then
+
+    trap 'cleanup' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts
+    curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    sudo cp "gh_${GH_CLI_VERSION}_linux_amd64/bin/gh" /usr/local/bin/
+    if ! command -V gh; then
+        echo "gh cli not installed successfully"
+        exit 1
+    fi
+fi
+
+./automation/travisci-test.sh
+
+# used in make targets
+# shellcheck disable=SC2034
+DOCKER_TAG="$tag"
+make manifests
+make build-functests
+
+# add artifacts to created github release
+set +u
+GITHUB_TOKEN="${GITHUB_TOKEN:-$(<"$GITHUB_TOKEN_PATH")}" gh release upload _out/cmd/virtctl/virtctl-v* _out/manifests/release/demo-content.yaml _out/manifests/release/kubevirt-operator.yaml _out/manifests/release/kubevirt-cr.yaml _out/templates/manifests/release/kubevirt-operator.yaml.j2 _out/templates/manifests/release/kubevirt-cr.yaml.j2 _out/manifests/release/olm/kubevirt-operatorsource.yaml "_out/manifests/release/olm/bundle/kubevirtoperator.$tag.clusterserviceversion.yaml" _out/tests/tests.test _out/manifests/release/conformance.yaml _out/manifests/release/conformance.yaml.j2
```

Comment:
Maybe we can use newlines for each file which we upload.

Reply:
Will we potentially leak the token here?

Reply:
This should probably also behind the dry-run flag.

Reply:
Not sure, I used this for testing the gh upload locally